var/spack/repos/builtin/packages/icu4c/package.py

```diff
@@ -39,11 +40,25 @@ class Icu4c(AutotoolsPackage):
     version('58.2', 'fac212b32b7ec7ab007a12dff1f3aea1')
     version('57.1', '976734806026a4ef8bdd17937c8898b9')
 
+    # The --enable-rpath option is needed on MacOS, but it breaks the
+    # build for xerces-c on Linux.
+    variant('rpath', default=sys.platform == 'darwin',
+            description='Configure with --enable-rpath. This is '
+            'needed for Darwin but should be avoided on Linux.')
+
+    conflicts('~rpath', when='platform=darwin')
+    conflicts('+rpath', when='platform=linux')
+
     configure_directory = 'source'
 
     def url_for_version(self, version):
         url = "http://download.icu-project.org/files/icu4c/{0}/icu4c-{1}-src.tgz"
         return url.format(version.dotted, version.underscored)
 
     def configure_args(self):
-        return ['--enable-rpath']
+        args = []
+
+        if '+rpath' in self.spec:
```

Comment:
@mwkrentel instead of making a `+rpath` variant, would it make sense to just check if `platform=darwin` here and add the `--enable-rpath` option then?

It looks like we always want RPATHs, but on MacOS we have to do some extra work for it.